bitcart/coins/btc.py

```diff
@@ -108,17 +112,18 @@ async def addrequest(
 
         Args:
             self (BTC): self
-            amount (Optional[Union[int, str]]): amount to open invoice. Defaults to None.
+            amount (Optional[AmountType]): amount to open invoice. Defaults to None.
             description (str, optional): Description of invoice. Defaults to "".
             expire (Union[int, float], optional): The time invoice will expire in. Defaults to 15.
 
         Returns:
             dict: Invoice data
         """
         expiration = 60 * expire if expire else None
-        data = await self.server.add_request(amount=amount, memo=description, expiration=expiration, force=True)
-        data[self.amount_field] = convert_amount_type(data[self.amount_field])
-        return data  # type: ignore
+        data = await self._addrequest(amount=amount, memo=description, expiration=expiration, force=True)
+        if data[self.amount_field] != "unknown":
```

Comment:
what does "unknown" mean?

Reply:
It is returned by electrum when amount is None.
Without this check it fails to convert to decimal